lib/Runtime/Language/JavascriptExceptionOperators.cpp

```diff
@@ -13,22 +13,30 @@ extern "C" PVOID __guard_check_icall_fptr;
 
 namespace Js
 {
-    void JavascriptExceptionOperators::AutoCatchHandlerExists::FetchNonUserCodeStatus(ScriptContext * scriptContext)
+    void JavascriptExceptionOperators::AutoCatchHandlerExists::FetchNonUserCodeStatus(ScriptContext * scriptContext, bool isUserExceptionHandling)
     {
         Assert(scriptContext);
 
         bool fFound = false;
         // If the outer try catch was already in the user code, no need to go any further.
         if (!m_previousCatchHandlerToUserCodeStatus)
         {
-            Js::JavascriptFunction* caller;
-            if (JavascriptStackWalker::GetCaller(&caller, scriptContext))
+            if (!isUserExceptionHandling)
             {
-                Js::FunctionBody *funcBody = NULL;
-                if (caller != NULL && (funcBody = caller->GetFunctionBody()) != NULL)
+                fFound = true;
```

Comment:
Thanks, I've removed all instances of fFound and used returns to replace the related logic in the new commit on the PR.